src/plugins/charts/bar/BarGraphView.vue

```diff
@@ -237,23 +287,40 @@ export default {
                 this.openmct.notifications.alert(data.message);
             }
 
-            if (!this.isDataInTimeRange(data, key)) {
+            if (!this.isDataInTimeRange(data, key, telemetryObject)) {
+                return;
+            }
+
+            if (this.domainObject.configuration.axes.xKey === undefined || this.domainObject.configuration.axes.yKey === undefined) {
                 return;
             }
 
             let xValues = [];
             let yValues = [];
+            const xAxisMetadata = axisMetadata.xAxisMetadata.find(metadata => metadata.source === this.domainObject.configuration.axes.xKey);
+            if (xAxisMetadata && xAxisMetadata.isArrayValue) {
+                //populate x and y values
+                let metadataKey = this.domainObject.configuration.axes.xKey;
+                if (data[metadataKey] !== undefined) {
+                    xValues = this.format(key, metadataKey, data);
```

Comment:
Hmm. I had used format since we need the human readable string for displaying the plot text. But I think I could use parse and then cast it to a string.